tests/model/test_model.py

```diff
@@ -1430,6 +1429,42 @@ def test_fetch_raw_message_content(
         assert self.display_error_if_present.called == display_error_called
         assert return_value == expected_raw_content
 
+    @pytest.mark.parametrize(
+        "response, expected_read_receipts, display_error_called",
+        [
+            (
+                {"msg": "", "result": "success", "user_ids": [3, 7, 9]},
+                "Verona, alice, Bob",
+                False,
+            ),
+            (
+                {"msg": "Invalid message(s)", "result": "error"},
+                None,
+                True,
+            ),
+        ],
+    )
+    def test_fetch_read_receipts(
+        self,
+        mocker,
+        model,
+        expected_read_receipts,
+        response,
+        display_error_called,
+        message_id=1,
+    ):
+        # self.client.fetch_read_receipts = mocker.patch(MODEL+".fetch_read_receipts", return_value="response")
+        result = model.fetch_read_receipts(message_id)
+        assert self.display_error_if_present.called == display_error_called
+        assert result == expected_read_receipts
```

Comment:
Let's refer to the code you're testing:
```python3
 response = self.client.call_endpoint(
            f"/messages/{message_id}/read_receipts", method="GET"
        )
 if response["result"] == "success":
     return ", ".join(list(map(self.user_name_from_id, response["user_ids"])))
```
This means the test tries to call the function `self.client.call_endpoint` in the model. However, the `model` you're using here is the model *fixture*, and if you check the client object in there, you'll see it's a mock (`MagicMock` actually). I'd suggest you try that in one of the model tests by print'ing out it and some other elements.

Since the client mock is not set up to respond a certain way to 'call_endpoint', it just creates it as it needs it. That's why the test behaves the same way in each parametrize case.

If you set it explicitly like in the test just below:
```python3
model.client.update_subscription_settings.return_value = response
```
but appropriately for call_endpoint, that should fix your issue.